lib/board.rb

```diff
@@ -1,33 +1,63 @@
 class Board 
-	attr_accessor :matrix
-	def initialize
-		@matrix = [nil, nil, nil, nil, nil, nil, nil, nil, nil]
+	attr_accessor :matrix, :win_combinations, :dim
+	def initialize(dim)
+		@matrix = Array.new(dim**2)
+		@win_combinations = init_combin(dim)
+		@dim = dim
+	end
+  
+  	def init_combin(dim)
+  		arr = []
+        arr  += Array.new(dim ){|l| Array.new(dim){|i| dim * l + i }}  
+        arr  += Array.new(dim){|l| Array.new(dim){|i|  l + (i*dim)}}
+        arr  += Array.new(1){|l| Array.new(dim){|i|  i+ (i*dim)}}
+        arr  += Array.new(1){|l| Array.new(dim){|i|  i*(dim-1) + (dim-1)}}
+        return arr                      
+    end
+
+    def valid_input?(arr)
+		if arr.length == 2 && 
+		(ALPHABET[0...@dim].any? {|val| val == arr[1].upcase}) &&
+		((0...@dim).to_a.any? {|element| element == arr[0].to_i})
+			return true
+		end
+		false
+	end
+
+	def empty_space?(i)
+		return true if @matrix[i] == nil
+		false
+	end
```

Comment:
Exactly what I meant with 'half validation'. 
Choose one strategy:
- Validate the input on `Game` (outside the `board` object) and then pass the input to the board
- Validate input on `board` (Different methods for validation maybe?) and run the validators before a move is made. Then `Game` just has to call board.make_move (or similar). [I prefer this strategy but is up to you]